plugins/action_send_notification/src/sendNotification.ts

```diff
@@ -80,14 +104,20 @@ const sendNotification: ActionPluginType = async ({ parameters, applicationData,
     })
 
     // Send email
-    if (sendEmail && hasValidEmails && transporter && !suppressEmail) {
-      console.log('Sending email...')
+    console.log(`Email mode: ${mode}`)
+    console.log(`Action sendEmail setting: ${sendEmail}`)
+    if (mode !== 'NO_EMAIL' && sendEmail && hasValidEmails && transporter) {
+      console.log(
+        `Sending email to: ${toAddressString}\ncc:${ccAddressString}\nbcc: ${bccAddressString}\nSubject: ${
+          subject || ''
+        }\n`
+      )
       transporter
         .sendMail({
           from: `${fromName} <${fromEmail}>`,
-          to: toAddressString,
-          cc: ccAddressString,
-          bcc: bccAddressString,
+          to: mode === 'TEST_EMAIL' ? '' : toAddressString,
+          cc: mode === 'TEST_EMAIL' ? '' : ccAddressString,
+          bcc: mode === 'TEST_EMAIL' ? testingEmail : bccAddressString,
```

Comment:
The testing email is sent as "bcc", I guess just to make it slightly more clear to the recipient that they're not necessarily the intended recipient. But this is somewhat dubious logic, so I'm happy to just make it the `to` field if that seems clearer?